### ISSUE TITLE
[IEEE 802.1Qci PSFP] Unknown Protocol in TSNSwitch (Simple)Ieee8021qFilter

### ISSUE THREAD

**Issue by olfline** [inet-framework/inet#855]
Hello dears,

I am encountering a problem concerning the (Simple)Ieee8021qFilter Compound Module in the TSN-Switch.

**SETUP:**
I have the following setup with the [ini.txt](https://github.com/inet-framework/inet/files/10709735/ini.txt) and [ned.txt](https://github.com/inet-framework/inet/files/10709736/ned.txt) files.
![network](https://user-images.githubusercontent.com/50140648/218152906-58f8797a-669a-4bb7-a97e-cbb5cac14163.png)

**PROBLEM:**
I'm getting the follwing runtime error when starting the simulation

> <!> handlePacket(): Unknown protocol: protocolId = 38, protocolName = ipv4, servicePrimitive = INDICATION, pathStartGate = simpleStart.switch.ieee8021q.qTagHeaderChecker.out, pathEndGate = simpleStart.switch.bl.in[1] -- in module (inet::MessageDispatcher) simpleStart.switch.bl (id=157), at t=0.00000581s, event #&#8203;9

**CAUSE:**
I could track it down to the StreamFilter module. The ContentBasedFilter is causing the same error.
Exchanging the StreamFilter with the LabelFilter, the Simulation is working fine. 

The StreamFilter and the ContentBasedFilter do not forward the Protocol registration. Thats why the protocolToGate map in the submodule "bl" of the TSN switch is missing one entry.
You can see this in the logs.

LabelFilter (protocol registration is forwarded)
![ProtoForwarding](https://user-images.githubusercontent.com/50140648/218152984-fa0847b4-5812-44bf-ade0-d53e74100473.png)

ContentBasedFilter (protcol registration is NOT forwarded - it stops at the Multiplexer sm right before the Filter module)
![noProtoForwarding](https://user-images.githubusercontent.com/50140648/218153019-fe8bfa67-06a1-4aa2-bdaf-e459801734f2.png)

*Note:*
I get the same behaviour when using the **SimpleIeee8021Filter**, exchanging the LabelFilter and disable the default path.

Could you please verify?

Omnet++ V6.0.1
Inet V4.4.1 - Commit 3570aad

**Comment by levy** [comment]
I checked and I got the same error.

**Comment by levy** [comment]
Fixed this in the INET master branch by adding the TransparentProtocolRegistrationListener base class to the mentioned packet filters. Ideally the PacketFilterBase would already have the TransparentProtocolRegistrationListener as base class but that causes issues with circular protocol registration because protocol header checkers are also derive from PacketFilterBase. There is no easy way to solve that issue, hence I added the base classes individually to the filters.
